### ISSUE TITLE
Add a check to ensure reference track and cooler have the same resolution

### ISSUE THREAD

**Issue by cmdoret** [open2c/cooltools#180]
Hello, 

I just wanted to report what I think to be a bug in the CLI. When I use the option `--reference-track`, the compartments end up all positive and seem wrong (no correlation with gene expression). When I don't use the `--reference-track` option, the eigen vectors are not flipped / ranked, but they are fine.

When using the API directly via `cooler_cis_eig`, everything works well even when I provide a phasing track.

I encountered this issue with 2 separate dataset (mouse and human). Here is the head of the (tab-separated) reference-track file I used:

```
chrom   start   end     rna_coverage
chr1    0       10000   654.3748188
chr1    10000   20000   0
chr1    20000   30000   0
chr1    30000   40000   1445.995385
chr1    40000   50000   1418.11489
```

Versions:
* python: 3.7.8
* cooltools: 0.3.2
* pandas: 1.1.0
* numpy: 1.19.1

I'll be having a look at the code, but if you have any idea, that'd be great :)
Best

**Comment by cmdoret** [comment]
OK I think I figured it out !
It's because the resolution of my cool file and the interval size on the bedgraph was different.
The problem happens during the merge of the phasing and bins tables:
https://github.com/mirnylab/cooltools/blob/95e744f43581b62c8baba3bc3b74ceff13f8cdf3/cooltools/cli/call_compartments.py#L131

I would suggest to either raise an error if resolutions are different, or to merge only on chrom and start with a warning, maybe ?

**Comment by gfudenberg** [comment]
proposed function: `checks.cooler_bins_in_track()`
-- for each bin in the cooler bintable, check that it is present in the track
-- with possible exception of last bin on each chromosome

**Comment by gfudenberg** [comment]
update 12/08/2021:

requirements above may be too strict, and current implementations in saddle & eigs are not compatible. what we want is also not really a check, since it returns a modified track.

new proposal 
```python
lib.common.align_track_to_clr(track,
                                                    clr,
                                                    view_df=None,
                                                    clr_weight_name='weight',
                                                    mask_bad_bins=True,
):
```

- check lib.checks.is_track() (no overlaps, bedframe, numeric 4th column)
- check median track bin size, cooler bin size. raise error if disagreeing
- try to pd.merge on ['chrom', 'start']
  - raise warning if <50% of valid clr.bins are assigned a value
  - raise an error if no values assigned
  - raise an error if any region with valid bins is not assigned any values. regions defined either by `make_cooler_view` or passed view
- propagate bad bins from bintable if `mask_bad_bins=True`, if the cooler is balanced with `clr_weight_name`, otherwise raise an error.


this function could supersede mask_cooler_bad_bins, 
https://github.com/open2c/cooltools/blob/880a1487c66a404e88437fd697caf4cd91dc0004/cooltools/lib/common.py#L192

**Comment by gfudenberg** [comment]
two more points: 
- `merge_track_with_cooler` could be a better name than `align_track_to_clr`
- check for sorting within chromosomes should be added `to is_track`

**Comment by gfudenberg** [comment]
resolved with #320
